projects/adc_challenge/src/main.c

```diff
@@ -0,0 +1,64 @@
+#include "adc.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "wait.h"
+
+// reads from the specified address and logs the value
+static void prv_read_from_adc_addr(const GpioAddress *address, void *context) {
+  GpioAddress *addr = context;
+  uint16_t adc_data = 0;
+  adc_read_converted_pin(*addr, &adc_data);
```

Comment:
This function will actually return a value of type `StatusCode`
If we don't get `STATUS_CODE_OK` we know we are getting corrupted data.
At the moment, we don't check for this. We could be printing out garbage without us knowing it.

Though what you have here is okay for the homework, for production code, we should almost always check the StatusCodes to make sure thing's dont fail (unless we are certain they won't)